### PR TITLE
fix(runtime): Fix bugs regarding lua runtime files

### PR DESCRIPTION
**Pull request by shadmansaleh**
### 1. Fix inconsistent verbose message on sourcing lua 

When a keymap is set from lua currently verbose message always says
it's set from line 1.

Example

That's incorrect because we don't really know when
it was set. So until proper `:verbose` support isn't added for lua `:verbose` should's say the line no.

### 2. fix(source): Source giving E484 & parsing error at line 1 for lua files

It's happening because do_source is only expected to return FAIL when it
was unable to open file . But `nlua_exec_file` returns fail for parsing
and execution error too . Those errors are emitted through `nlua_error`.

So now return value of nlua_exec_file is ignored like do_cmdline. It now
only returns fail when it was unable to open file that check is done
before calling nlua_exec_file or do_cmdline. Errors in nlua_exec_file
are still directly emitted through nlua_error like before.

### CODE REVIEW

File: src/nvim/ex_cmds2.c [hunk 1]
```diff
@@ -3002,8 +3003,17 @@ int do_source(char_u *fname, int check_other, int is_vimrc)
   }
 
   if (path_with_extension((const char *)fname, "lua")) {
+    // TODO(shadmansaleh): Properly handle :verbose for lua
+    // For now change currennt_sctx before sourceing lua files
+    // So verbose desn't say everything was done in line 1 aince we don't know
```
Comment (bfredl) [inline src/nvim/ex_cmds2.c:3008]: aince -> since

**Comment by bfredl** [comment]
LGTM. getting lua line numbers should be possible using `debug`, let's aim for it in 0.6.

File: src/nvim/lua/executor.c [hunk 2]
```diff
@@ -1417,20 +1427,20 @@ void ex_luafile(exarg_T *const eap)
 ///
 /// @param  path  path of the file
 ///
-/// @return  true if everything ok, false if there was an error (echoed)
+/// @return  false when unable to loadfile othewise return true
```
Comment (seandewar) [inline src/nvim/lua/executor.c:1430]: LGTM, but maybe be explicit here that `false` is returned if opening or reading the file fails (to disambiguate it from the other things `loadfile` does).

File: src/nvim/ex_cmds2.c (hunk 1 above)
Comment (shadmansaleh) [reply src/nvim/ex_cmds2.c]: fixed

File: src/nvim/lua/executor.c (hunk 2 above)
Comment (shadmansaleh) [reply src/nvim/lua/executor.c]: good catch . I actually forgot to change the comment after changing the code . Fixed now :)

**Comment by shadmansaleh** [comment]
We don't need to change `nlua_exec_file`s behavior . We just should ignore it's return value in `do_source`. Because when file can't be opened neither `nlua_exec_file` nor `do_cmdline` gets called . Instead 

https://github.com/neovim/neovim/blob/b585f723bcbaa10a091fce5a61659b331b1467b9/src/nvim/ex_cmd2.c#L2865

jumps to the end . So we will still catch E484.

@bfredl this seems like a better solution . Let me know what you think.

> getting lua line numbers should be possible using debug, let's aim for it in 0.6.

I'll check how `debug` can be used for `:verbose` . By the way do you know what we actually need to cover . I'm aware of `:verbose map` and `:verbose set` is there anything else ?

**Comment by seandewar** [comment]
> We don't need to change `nlua_exec_file`s behavior . We just should ignore it's return value in `do_source`. Because when file can't be opened neither `nlua_exec_file` nor `do_cmdline` gets called . Instead 
> 
> ... 
> 
> will catch that file can't be opened and `do_source` will return FAIL . So we will still catch E484.

I was aware you could do that, but if `luaL_loadfile` was to then have issues reading the file for some reason I'd still expect callers to be notified (E484). Interested to see @bfredl 's opinion here. 

Tbh I guess the ideal solution would be to let `luaL_loadfile` handle reading the file itself at some point in the future (`do_source` is doing some amount of VimL-specific work that goes unused when sourcing Lua).

**Comment by shadmansaleh** [comment]
> I'd still expect callers to be notified (E484).

That would actually be E485 . It ins't addresed in `do_source` . 

> Tbh I guess the ideal solution would be to let luaL_loadfile handle reading the file itself at some point in the future (do_source is doing some amount of VimL-specific work that goes unused when sourcing Lua).

I agree.

**Comment by seandewar** [comment]
> > I'd still expect callers to be notified (E484).
> 
> That would actually be E485 . It ins't addresed in `do_source` . I don't think think `do_source` fails in that case for vimL either.
> 

That's true, but `luaL_loadfile` would still need to open the file and `LUA_ERRFILE` doesn't seem to differentiate the two. Seeing that no side effects from running the script would have happened in either case I think returning `FAIL` on a read error is preferable even if the error message is a bit off. 👍

**Pull request merged**